firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -252,10 +252,7 @@ class TabManagerImplementation: LegacyTabManager, Notifiable, WindowSimpleTabsPr
     }
 
     private func generateTabDataForSaving() -> [TabData] {
-        var tabsToSave = tabs
-        if shouldClearPrivateTabs() {
-            tabsToSave = normalTabs
-        }
+        let tabsToSave = tabs
```

Comment:
Thanks for digging into this @PARAIPAN9. 🙏 

> Therefore, if we change the option in settings and force crash the app, the default value remains true, and no changes are applied.

To me this seems like the root issue. Is the change to UserDefaults not persisting correctly? (Wondering if we need to add a call to `synchronize()` in order to force the UserDefaults to be updated to disk.)

>  let tabsToSave = tabs

Lmk if I'm missing something obvious, but when I test this change locally it looks like private tabs are being written out even if the setting is on? If so, that's problematic because we need to avoid writing private tabs to disk in that scenario (it's a privacy issue).

```
(lldb) po tabsToSave.map { $0.isPrivate}
▿ 2 elements
  - 0 : false
  - 1 : true     
(lldb) po shouldClearPrivateTabs()
true
```

Reply:
Maybe I didn't explain it very well, sorry. The shouldClearPrivateTabs() function gets updated when toggling the switch, but because we already filtered the tabs and the app does not enter the background to record this change(the app crashes), it remains in the old state. We could find another solution to avoid storing private tabs to disk, as you mentioned, since this is not what we want.

Reply:
Thanks for clarifying. 👍 

> We could find another solution to avoid storing private tabs to disk, as you mentioned, since this is not what we want.

AFAIU we can't write private tabs to disk if the setting is on (`shouldClearPrivateTabs` is true) because that would regress the privacy issue which the toggle was originally added to solve.